stackstac/rio_reader.py

```diff
@@ -399,14 +400,16 @@ def read(self, window: Window, **kwargs) -> np.ndarray:
 
             raise RuntimeError(msg) from e
 
-        if self.rescale:
-            scale, offset = reader.scale_offset
-            if scale != 1:
-                result *= scale
-            if offset != 0:
-                result += offset
+        if result.dtype != self.dtype:
+            result = result.astype(self.dtype, copy=False)
```

Comment:
Note that this becomes unnecessary after https://github.com/gjoseph92/stackstac/pull/208, so depending on merge order, it can be removed. (This is actually just a different way of solving #206.)

Reply:
I am happy to remove and delegate to #208 as it is authored by you. What is your timeline for merging that?

Reply:
I was hoping to hear confirmation from @Berhinj that it solves the problem before merging, since I don't have a reproducer. However, I'm quite confident it'll work, so I'll just merge it now so we can get these both in.

I think I'd slightly prefer that approach to what you have here, just because it _maybe_ saves one copy (letting GDAL read the data into an array of the desired output dtype, vs copying into a new array). There are so many copies internally already though, I doubt it matters much.